crates/client/eth/src/state_update.rs

```diff
@@ -30,85 +26,65 @@ pub async fn get_initial_state(client: &EthereumClient) -> anyhow::Result<L1Stat
-        )?
-        .into_stream();
-
-    while let Some(event_result) = channel_wait_or_graceful_shutdown(event_stream.next(), &ctx).await {
-        let log = event_result.context("listening for events")?;
-        let format_event: L1StateUpdate =
-            convert_log_state_update(log.0.clone()).context("formatting event into an L1StateUpdate")?;
-        update_l1(backend, format_event, block_metrics, chain_id.clone())?;
-    }
-
-    Ok(())
-}
-
 pub fn update_l1(
     backend: &MadaraBackend,
     state_update: L1StateUpdate,
     block_metrics: &L1BlockMetrics,
-    chain_id: ChainId,
 ) -> anyhow::Result<()> {
-    // This is a provisory check to avoid updating the state with an L1StateUpdate that should not have been detected
-    //
-    // TODO: Remove this check when the L1StateUpdate is properly verified
-    if state_update.block_number > 500000u64 || chain_id.to_felt() == MAIN_CHAIN_ID {
-        tracing::info!(
-            "🔄 Updated L1 head #{} ({}) with state root ({})",
-            state_update.block_number,
-            trim_hash(&state_update.block_hash),
-            trim_hash(&state_update.global_root)
-        );
+    tracing::info!(
+        "🔄 Updated L1 head #{} ({}) with state root ({})",
+        state_update.block_number,
+        trim_hash(&state_update.block_hash),
+        trim_hash(&state_update.global_root)
+    );
 
-        block_metrics.l1_block_number.record(state_update.block_number, &[]);
+    block_metrics.l1_block_number.record(state_update.block_number, &[]);
 
-        backend
-            .write_last_confirmed_block(state_update.block_number)
-            .context("Setting l1 last confirmed block number")?;
-        tracing::debug!("update_l1: wrote last confirmed block number");
-    }
+    backend.write_last_confirmed_block(state_update.block_number).context("Setting l1 last confirmed block number")?;
+    tracing::debug!("update_l1: wrote last confirmed block number");
 
     Ok(())
 }
 
 pub async fn state_update_worker(
     backend: &MadaraBackend,
     eth_client: &EthereumClient,
-    chain_id: ChainId,
-    ctx: ServiceContext,
+    mut ctx: ServiceContext,
 ) -> anyhow::Result<()> {
     // Clear L1 confirmed block at startup
     backend.clear_last_confirmed_block().context("Clearing l1 last confirmed block number")?;
     tracing::debug!("update_l1: cleared confirmed block number");
 
     tracing::info!("🚀 Subscribed to L1 state verification");
-    // ideally here there would be one service which will update the l1 gas prices and another one for messages and one that's already present is state update
-    // Get and store the latest verified state
-    let initial_state = get_initial_state(eth_client).await.context("Getting initial ethereum state")?;
-    update_l1(backend, initial_state, &eth_client.l1_block_metrics, chain_id.clone())?;
+    // This does not seem to play well with anvil
+    #[cfg(not(test))]
```

Comment:
that's a hack? why is it here?

Reply:
I refactored this from the previous version which separated `state_update_worker` into `listen_and_update_state`. I did not see the reason for that and so included it directly into `state_update_worker`. For some reason I don't quite understand though, getting the initial l1 state does not work, which might could either be an issue in the way in which we are mocking the core contract or with how we are using anvil.